### PR TITLE
Serializer fixes

### PR DESCRIPTION
**Pull request by tristantarrant**
https://issues.jboss.org/browse/ISPN-6646
https://issues.jboss.org/browse/ISPN-6653
https://issues.jboss.org/browse/ISPN-6654
https://issues.jboss.org/browse/ISPN-6655
https://issues.jboss.org/browse/ISPN-6656
https://issues.jboss.org/browse/ISPN-6657
https://issues.jboss.org/browse/ISPN-6658
https://issues.jboss.org/browse/ISPN-6659
https://issues.jboss.org/browse/ISPN-6660
https://issues.jboss.org/browse/ISPN-6661
https://issues.jboss.org/browse/ISPN-6662
https://issues.jboss.org/browse/ISPN-6687
https://issues.jboss.org/browse/ISPN-6690
https://issues.jboss.org/browse/ISPN-6701

### CODE REVIEW

**Comment by tristantarrant** [comment]
Neither of the failures are related

File: core/src/test/resources/configs/unified/9.0.xml [hunk 1]
```diff
@@ -49,7 +49,7 @@
          <expiration interval="10000" lifespan="10" max-idle="10"/>
          <persistence passivation="false">
             <file-store path="path" relative-to="jboss.server.temp.dir" shared="true" singleton="false" fetch-state="false" preload="true" purge="false">
-               <write-behind flush-lock-timeout="2" modification-queue-size="2048" shutdown-timeout="20000" thread-pool-size="1" />
+               <write-behind modification-queue-size="2048" thread-pool-size="1" />
```
Comment (pruivo) [inline core/src/test/resources/configs/unified/9.0.xml]: why removing this attributes?

Comment (pruivo) [inline core/src/test/resources/configs/unified/9.0.xml]: nevermind :)

Comment (tristantarrant) [inline core/src/test/resources/configs/unified/9.0.xml]: They don't do anything anymore.

File: core/src/main/java/org/infinispan/configuration/parsing/Parser.java [hunk 2]
```diff
@@ -54,6 +54,7 @@
 import org.infinispan.security.impl.CommonNameRoleMapper;
 import org.infinispan.security.impl.IdentityRoleMapper;
 import org.infinispan.transaction.LockingMode;
+import org.infinispan.transaction.TransactionMode;
```
Comment (pruivo) [inline core/src/main/java/org/infinispan/configuration/parsing/Parser.java]: unused import

File: core/src/main/java/org/infinispan/configuration/serializing/Serializer.java [hunk 3]
```diff
@@ -131,11 +135,52 @@ private void writeThreadFactory(XMLExtendedStreamWriter writer, DefaultThreadFac
...
+            DefaultThreadFactory tf = (DefaultThreadFactory) threadFactory;
+            writer.writeAttribute(Attribute.THREAD_FACTORY, tf.getName());
+         }
+         if (threadPoolFactory instanceof BlockingThreadPoolExecutorFactory) {
+            BlockingThreadPoolExecutorFactory pool = (BlockingThreadPoolExecutorFactory) threadPoolFactory;
+            writer.writeAttribute(Attribute.MAX_THREADS, Integer.toString(pool.maxThreads()));
+            writer.writeAttribute(Attribute.CORE_THREADS, Integer.toString(pool.coreThreads()));
+            writer.writeAttribute(Attribute.QUEUE_LENGTH, Integer.toString(pool.queueLength()));
+            writer.writeAttribute(Attribute.KEEP_ALIVE_TIME, Long.toString(pool.keepAlive()));
+         } else if (threadPoolFactory instanceof CachedThreadPoolExecutorFactory) {
```
Comment (pruivo) [inline core/src/main/java/org/infinispan/configuration/serializing/Serializer.java]: this else branch is not really needed.

File: server/integration/infinispan/src/main/java/org/jboss/as/clustering/infinispan/subsystem/StoreWriteBehindResource.java [hunk 4]
```diff
@@ -23,6 +23,7 @@
 package org.jboss.as.clustering.infinispan.subsystem;
 
 import org.jboss.as.controller.AttributeDefinition;
+import org.jboss.as.controller.ModelVersion;
```
Comment (pruivo) [inline server/integration/infinispan/src/main/java/org/jboss/as/clustering/infinispan/subsystem/StoreWriteBehindResource.java]: unused import

File: core/src/main/java/org/infinispan/configuration/cache/TransactionConfiguration.java [hunk 5]
```diff
@@ -23,19 +23,19 @@
...
    public static final AttributeDefinition<Boolean> USE_1_PC_FOR_AUTO_COMMIT_TRANSACTIONS = AttributeDefinition.builder("single-phase-auto-commit", false).build();
-   public static final AttributeDefinition<Long> REAPER_WAKE_UP_INTERVAL = AttributeDefinition.builder("reaper-wake-up-interval", 30000l).immutable().build();
+   public static final AttributeDefinition<Long> REAPER_WAKE_UP_INTERVAL = AttributeDefinition.builder("reaper-wake-up-interval", 30000l).immutable().xmlName("reaper-interval").build();
    public static final AttributeDefinition<Long> COMPLETED_TX_TIMEOUT = AttributeDefinition.builder("complete-timeout", 60000l).immutable().build();
-   public static final AttributeDefinition<TransactionProtocol> TRANSACTION_PROTOCOL = AttributeDefinition.builder("transaction-protocol", TransactionProtocol.DEFAULT).immutable().build();
+   public static final AttributeDefinition<TransactionProtocol> TRANSACTION_PROTOCOL = AttributeDefinition.builder("transaction-protocol", TransactionProtocol.DEFAULT).immutable().autoPersist(false).build();
```
Comment (pruivo) [inline core/src/main/java/org/infinispan/configuration/cache/TransactionConfiguration.java]: this is not serialized and it should be under the xml name ```protocol```

**Comment by pruivo** [comment]
@tristantarrant you forgot one comment: https://github.com/infinispan/infinispan/pull/4355#discussion_r64391179

**Comment by tristantarrant** [comment]
Done @pruivo

**Comment by pruivo** [comment]
pushing...

**Comment by pruivo** [comment]
integrated! thanks @tristantarrant !

**Pull request closed without merging**